### ISSUE TITLE
Sessions for 2019-20

### ISSUE THREAD

**Issue by dnabanita7** [p-society/goals#49]
- The chronological order of sessions for PSoc as suggested on `nullptr` are:
1. Introduction to Competitive Programming (will plan in detail) @sibasish14 @PRIYANSUPULAK @niket @PritishN 
2. Python : give them basic resources to learn and start implementing those on some pet projects (1.5 hrs for 2 days)  @piyushmishra12 @nebula001 @dax-1895 @sundarayamrita  @rkt-tryme @Naba7
3. Since, after DP everyone would have their systems, we can organize a Linux session : (1.5 hrs for 3 days) @KayGKay @Swaraj1998 @MetalDent 
4. OSS session : intro to distributed version-control, intro to Git, GitHub, gitlab, other subsidiaries, intro to Open source Program, basics of Git-flows (1.5 hrs for 3 days) @sudonitesh @debck @Rupeshiya @Naba7 
5. WebD/Android : Giving them resources and doing a pet project (1.5 hrs for 3 days) @debck @quoraboy @pranayag15 @suicide11 @pratap47 @MetalDent @nayonika1999 @Mownika25 @sofia100
6. ML : basics of statistics, probability for this semester and introduction to kaggle #42 competitions, kernels, a regression/classification competition problem such as House Price prediction to be solved @piyushmishra12 @nebula001  @sundarayamrita @rkt-tryme @Naba7

- A competitive coding competition during vacations (both Durga Puja and winter) @sibasish14 @niket @PRIYANSUPULAK 

- We are planning for a Winter Of Code this time. Freshmen and Sophomore years can participate. So, as they are introduced to some of the domains Freshmen years can participate too and **We need more projects so that they can work on/ any oss organisation interested to participate.**

- So, we want to stretch these sessions over two semesters rather than a single semester. The target is to complete the basics in this semester for all the sessions and go for advanced sessions next semester.

For the even semester, the plan is as follows:
1. Competitive coding: (not sure what exactly, will decide as per the requirements)
2. Go(as suggested by 2nd years in the first meeting): @KayGKay 
3. WebD/Android : 
4. ML/DL: Intro to ML/DL (solving a pet project), intro to text, vision, speech
5. Hackathons: @sudonitesh @Rupeshiya @quoraboy @debck @pratap47 @pranayag15 @suicide11 

- Technical Writing: Write for all the sessions conducted @sundarayamrita @snehastar @anwesha-mohanty 

All 2nd years get in contact with 3rd years for organizing the events.
Edit this to include whoever you want.

**Comment by palash25** [comment]
Hello :wave: 

>  GitHub, gitlab, other subsidiaries

imo introducing too many cloud hosted to the first years might be too much of information, so I hope by this you mean that you will introduce them to one of them (github would be my recommendation) and just tell the names of the other services

> intro to distributed version-control, intro to Git, GitHub, gitlab, other subsidiaries, intro to Open source Program, basics of Git-flows (1.5 hrs for 3 days)

so this was something that I wanted to do last year but couldn't and its also missing from this year's description. I wanted to show them the process of solving an issue. Turning something like [this](https://gist.github.com/palash25/253be11ad73380b44134ff2e98e97a58) into a live demo. I would be forever grateful if you all are able to do this.

> Go(as suggested by 2nd years in the first meeting)

who is gonna take this one?

**Comment by dnabanita7** [comment]
Hi
>  so I hope by this you mean that you will introduce them to one of them (github would be my recommendation) and just tell the names of the other services

Yes, I meant the same. Introducing them to every other cloud based git services in 1st year will confuse them and make them less motivated.

>  I wanted to show them the process of solving an issue.

By introducing to GitHub I mean not just explaining the terminologies used but making them raise an issue and create and merge a pr at the same time. We will make all the sessions more of hands-on rather than just explaining the theory. Most probably I think of using firstcontributions repo for this.

> I would be forever grateful if you all are able to do this.

We also want to do this. :smile: It might help increase number of GSoCers or other open source program students. 

> who is gonna take this one?

No idea. ping @debck 

Also, can the 4th years tell any problems or loopholes or shortcomings in conducting sessions so that we won't repeat that again.

**Comment by palash25** [comment]
> Most probably I think of using firstcontributions repo for this.

you can use that, but that's not what I meant. I wanted someone to show them how to go about making a PR for an actual good first issue in an actual OSS projects, not just creating PRs in their own repos or forks. If that is possible.

**Comment by kaygdotorg** [comment]
> who is gonna take this one?

I can. 

> github would be my recommendation

GitLab would be mine. It's completely OSS unlike GitHub.

> you can use that, but that's not what I meant. I wanted someone to show them how to go about making a PR for an actual good first issue in an actual OSS projects, not just creating PRs in their own repos or forks. If that is possible.

I guess this is possible since a lot of projects have very trivial issues that people keep deferring..

**Comment by palash25** [comment]
> GitLab would be mine. It's completely OSS unlike GitHub.

Agreed I just find github's interface easier to grasp unlike gitlab which looks too detailed and cluttered to me, anyway you all can take a call on this


> I guess this is possible since a lot of projects have very trivial issues that people keep deferring..

it could be something as small as this PR I made a few days ago https://github.com/nim-lang/Nim/pull/12000 An issue where you can tell them how to look for the related files, finding the relevant piece of code using something like `grep` and making the fix

**Comment by kaygdotorg** [comment]
> finding the relevant piece of code using something like grep and making the fix

Oh, I think that's a brilliant approach. It gives them insight on filtering too. 
And mate, when did you learn Nim?

**Comment by palash25** [comment]
> And mate, when did you learn Nim?

I didn't just been wanting to, but the official site has a plethora of resources to learn from so maybe I will if I get the time.

**Comment by dnabanita7** [comment]
>  I wanted someone to show them how to go about making a PR for an actual good first issue in an actual OSS projects, not just creating PRs in their own repos or forks. If that is possible.

Yes sure.

Well I also like GitLab rather than GitHub. But considering the fact that freshmen year is introduced, GitHub will be better. Maybe we need a discussion on this with others.
